packages/gamut/src/ConnectedForm/utils.tsx

```diff
@@ -0,0 +1,145 @@
+import { useContext } from 'react';
+import { RegisterOptions, useFormContext } from 'react-hook-form';
+
+import {
+  ConnectedForm,
+  ConnectedFormGroup,
+  ConnectedFormGroupProps,
+  ConnectedFormProps,
+  FormPropsContext,
+} from '.';
+import { SubmitContextProps } from './SubmitButton';
+import { ConnectedField } from './types';
+
+interface ConnectedGroupStrictProps<Values extends {}> {
+  <Name extends keyof Values, Component extends ConnectedField>(
+    props: {
+      name: Name;
+    } & ConnectedFormGroupProps<Component>
+  ): ReturnType<typeof ConnectedFormGroup>;
+}
+
+interface UseConnectedFormProps<
+  Values,
+  Rules extends { [Key in keyof Values]?: RegisterOptions }
+> {
+  defaultValues: Values;
+  validationRules: Partial<Rules>;
+}
+
+interface ConnectedFormStrictProps<Values, Rules>
+  extends UseConnectedFormProps<Values, Rules> {
+  (
+    props: UseConnectedFormProps<Values, Rules> & ConnectedFormProps<Values>
+  ): ReturnType<typeof ConnectedForm>;
+}
+
+export const useConnectedForm = <
+  Values,
+  ValidationRules extends { [K in keyof Values]: RegisterOptions }
+>({
+  defaultValues,
+  validationRules,
+}: UseConnectedFormProps<Values, ValidationRules>) => {
+  return {
+    ConnectedFormGroup: ConnectedFormGroup as ConnectedGroupStrictProps<Values>,
+    ConnectedForm: ConnectedForm as ConnectedFormStrictProps<
+      Values,
+      ValidationRules
+    >,
+    connectedFormProps: {
+      defaultValues,
+      validationRules,
+    },
+  };
```

Comment:
We'll need to override the lint rule here, but I think this what we want to do to prevent constant context updates:
```suggestion
  return useMemo(() => ({
    ConnectedFormGroup: ConnectedFormGroup as ConnectedGroupStrictProps<Values>,
    ConnectedForm: ConnectedForm as ConnectedFormStrictProps<
      Values,
      ValidationRules
    >,
    connectedFormProps: {
      defaultValues,
      validationRules,
    },
  }), []);
```

Can you think of cases where validationRules / default values are changed?  What we could do is only call it if the form has been reset and keep track of state in the hook?   Passing the callback down to trigger a recompute :).